tools/distrib/generate_grpc_shadow_boringssl.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+# Copyright 2018 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script generates a list of preprocessor macros to overwrite BoringSSL
+# symbol names by adding GRPC_SHADOW prefix. This is necessary to make gRPC
+# work with apps that also has OpenSSL dependency on iOS.
+# The list of symbols to be overwritten is in
+# src/objective-c/grpc_shadow_boringssl_symbol_list. This script generates the
+# macros in src/core/tsi/grpc_shadow_boringssl for gRPC core and
+# src/objective-c/BoringSSL.podspec for BoringSSL.
+set -e
+
+cd $(dirname $0)
+
+filename_original="../../src/core/tsi/grpc_shadow_boringssl.h"
+boringssl_podspec_original="../../src/objective-c/BoringSSL-GRPC.podspec"
+if [ "$TEST" == "" ]; then
+  filename=$filename_original
+  boringssl_podspec=$boringssl_podspec_original
+else
+  filename="${filename_original}.tmp"
+  boringssl_podspec="${boringssl_podspec_original}.tmp"
+fi
+symbol_list="../../src/objective-c/grpc_shadow_boringssl_symbol_list"
+
+symbols=($(cat $symbol_list | tail -n +3))
+
+> $filename
+echo '/*' >> $filename
```

Comment:
I thinks you can whitelist the file in license checker so that license header is not required:
https://github.com/grpc/grpc/blob/82bc60c0e13bfb00213b3a94ba72893d044e4c9a/tools/distrib/check_copyright.py#L68

I think it's worth doing that as it would simplify the script.